src/robusta/core/triggers/container_oom_killed_trigger.py

```diff
@@ -0,0 +1,50 @@
+from typing import List
+
+from hikaru.model import Pod
+
+from .oom_killed_trigger_base import OOMKilledTriggerBase, Exclude
+from ...core.playbooks.base_trigger import TriggerEvent
+
+
+class ContainerOOMKilledTrigger(OOMKilledTriggerBase):
+    def __init__(
+            self,
+            name_prefix: str = None,
+            namespace_prefix: str = None,
+            labels_selector: str = None,
+            rate_limit: int = 0,
+            exclude: List[Exclude] = None
+    ):
+        super().__init__(
+            name_prefix=name_prefix,
+            namespace_prefix=namespace_prefix,
+            labels_selector=labels_selector,
+            rate_limit=rate_limit,
+            exclude=exclude,
+        )
+
+    def should_fire(self, event: TriggerEvent, playbook_id: str):
+        return super().should_fire(event, playbook_id)
+
+    def get_relevant_oomkilled_containers(self, pod: Pod):
+        containers = pod.status.containerStatuses + pod.status.initContainerStatuses
+        if not self.exclude:
+            return containers
+        for selector in self.exclude:
+            namespace = None if "namespace" not in selector else selector["namespace"]
+            name = None if "name" not in selector else selector["name"]
+
+            if not namespace and not name:
+                # bad config
+                continue
+            namespace_match = namespace and namespace == pod.metadata.namespace
+            if namespace and not namespace_match:
+                # this selector isnt the current namespace
+                continue
+            if namespace_match and not name:
+                # this selector is for all containers on this namespace
+                return False
+            # either namespace_match or namespace isnt defined for this name
+            containers = [container for container in containers if not container.name.startswith(name)]
+        # pod not excluded
+        return containers
```

Comment:
I think you can replace it all with the following, I think it's much clearer:

    def get_relevant_oomkilled_containers(self, pod: Pod) -> List[ContainerStatus]:
        statuses = pod.status.containerStatuses + pod.status.initContainerStatuses
        return [status for status in statuses 
                if not self.__is_container_excluded(status, pod.metadata.namespace)]

    def __is_container_excluded(self, container: ContainerStatus, namespace: str) -> bool:
         
        for selector in self.exclude:
            if self.__is_excluded(selector.namespace, namespace, False) and \
                    self.__is_excluded(selector.name, container.name, True):
                return True
        return False
                 
    @staticmethod
    def __is_excluded(exclude: str, value: str, prefix: bool) -> bool:
        if not exclude:
            return False
        if prefix:
            return value.startswith(exclude)
        else:
            return value == exclude